ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1347,6 +1347,14 @@ private String resolveReservedMetadataId() {
       if (!routerConfig.routerReservedMetadataEnabled || isSimpleBlob || isMetadataChunk()) {
         return null;
       }
+      // S3 doesn't use reserved metadata chunk.
+      // 1. For multipart upload, S3 doesn't use signed url. And for the data chunk, we don't reserve metadata chunk.
+      //    Customer may upload part by part for days.
+      // 2. For regular update, it's not chunked upload. It's uploaded with single PutObject.
+      //    Don't reserve metadata chunk either. We manage the S3 life cycle right away.
+      if (RestUtils.isS3Request(restRequest)) {
+        return null;
+      }
```

Comment:
let's use PutBlobOptions.skipCompositeChunk() here. Checking if the request is s3 is tying the feature too much with the s3.